test/routes/mailboxesAdminTest.js

```diff
@@ -0,0 +1,246 @@
+          expect(configClient.updateMailbox).to.be.calledWith('1', {
+            establishment: '1',
+            email: 'email1',
+            role: 'CA',
+            name: 'name1',
+          })
+        })
+    })
+  })
+
+  describe('GET /admin/mailboxes/delete', () => {
+    it('calls configClient and shows mailbox details', () => {
+      const app = createApp(configClient, 'batchUser')
+      return request(app)
+        .get('/admin/mailboxes/delete/1')
+        .expect(200)
+        .expect('Content-Type', /html/)
+        .expect(res => {
+          expect(configClient.getMailbox).to.be.calledOnce()
+          expect(configClient.getMailbox).to.be.calledWith('1')
+          expect(res.text).to.contain('<td>establishment1')
+          expect(res.text).to.contain('<td>role1')
+          expect(res.text).to.contain('<td>email1')
+          expect(res.text).to.contain('<td>name1')
+        })
+    })
+  })
+
+  describe('POST /admin/mailboxes/delete', () => {
+    it('calls configClient and redirects to mailbox list', () => {
+      const app = createApp(configClient, 'batchUser')
+      return request(app)
+        .post('/admin/mailboxes/delete/1')
+        .send()
+        .expect(302)
+        .expect('Location', '/admin/mailboxes')
+        .expect(() => {
+          expect(configClient.deleteMailbox).to.be.calledOnce()
+          expect(configClient.deleteMailbox).to.be.calledWith('1')
+        })
+    })
+  })
+
+  describe('GET /admin/mailboxes/add', () => {
+    const app = createApp(configClient, 'batchUser')
+    it('shows add mailbox form', () => {
+      return request(app)
+        .get('/admin/mailboxes/add/')
+        .expect(200)
+        .expect('Content-Type', /html/)
+        .expect(res => {
+          expect(res.text).to.contain('Add Mailbox')
+        })
+    })
+  })
+
+  describe('POST /admin/mailboxes/add', () => {
+    it('redirects back to page and does not call config client when missing email', () => {
+      const app = createApp(configClient, 'batchUser')
+      return request(app)
+        .post('/admin/mailboxes/add/')
+        .send({ establishment: 'establishment1', email: '', role: 'role1', name: 'name1' })
+        .expect(302)
+        .expect('Location', '/admin/mailboxes/add')
+        .expect(() => {
+          expect(configClient.addMailbox).not.to.be.calledOnce()
+        })
+    })
+
+    it('redirects back to page and does not call config client when missing establishment', () => {
+      const app = createApp(configClient, 'batchUser')
+      return request(app)
+        .post('/admin/mailboxes/add/')
+        .send({ establishment: '', email: 'email1', role: 'role1', name: 'name1' })
+        .expect(302)
+        .expect('Location', '/admin/mailboxes/add')
+        .expect(() => {
+          expect(configClient.addMailbox).not.to.be.calledOnce()
+        })
+    })
+
+    it('redirects back to page and does not call config client when role not CA or DM', () => {
```

Comment:
These tests are using a limited data set to try the features out. Might benefit from a wider range of data?

Reply:
Added some cases to make more explicit that any input is allowed regardless of format as long as mandatory fields present